Shazam.php

```diff
@@ -36,6 +33,145 @@ public function __construct()
         // if ($project_id) $this->loadConfig();
     }
 
+
+    /**
+     * Returns all users saved in project config
+     * @param void
+     * @return $js_users : array
+     */
+    public function getJavascriptUsers(){
+        if($this->getSystemSetting('enable-add-user-javascript-permissions')){
+            $js_users = json_decode($this->getProjectSetting(self::JS_USER_KEY));
+            return isset($js_users) ? $js_users : [];
+        }
+    }
+
+
+    /**
+     * Adds $username to project settings, enabling JS editing
+     * @param $username
+     * @return void
+     */
+    public function addJavascriptUser($username) {
+        if(SUPER_USER && $this->getSystemSetting('enable-add-user-javascript-permissions')){ //Check if user has permission to add
+            $existing_users = $this->getJavascriptUsers();
+
+            if(empty($existing_users)){ //if empty simply add to the list
+                $this->emDebug("No existing users, setting ", $username);
+                $this->setProjectSetting(self::JS_USER_KEY, json_encode(array($username)));
+                $this->sendNotificationEmail($username);
+            }else{ //check if username is within existing users already added
+                if(in_array($username, $existing_users)){
+                    $this->emDebug("Error, username {$username} exists within existing users, update choices");
+                } else {
+                    array_push($existing_users, $username);
+                    $this->setProjectSetting(self::JS_USER_KEY, json_encode($existing_users));
+                    $this->sendNotificationEmail($username);
+                }
+            }
+        } else {
+            $this->emError("User ${username} attempting to enable dynamic js permissions and failed");
+        }
+    }
+
+
+    /**
+     * Removes username from project settings
+     * @param $username
+     * @return void
+     */
+    public function removeJavascriptUser($username){
+        if(SUPER_USER && $this->getSystemSetting('enable-add-user-javascript-permissions')){ //check if user has permissions to remove
+            $existing_users = $this->getJavascriptUsers();
+
+            $index = array_search($username, $existing_users);
+            if(isset($index)){
+                unset($existing_users[$index]);
+                $re_index = array_values($existing_users); //necessary to prevent conversion to object upon encoding
+                $this->setProjectSetting(self::JS_USER_KEY, json_encode($re_index));
```

Comment:
I would probably break this out into a function:
public function saveJavaScriptUsers($users) {} 
and then call it from both the 'remove' and the 'add' functions.  This way you don't have the same/similar code in two places.